### PR TITLE
build: specify go version in govulncheck action

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Specify go version in govulncheck action to stop errors like this https://github.com/kanisterio/kanister/actions/runs/9105298077/job/25030632704

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by hairyhum** [comment]
Test run: https://github.com/kanisterio/kanister/actions/runs/9116039558

File: .github/workflows/govulncheck.yaml [hunk 1]
```diff
@@ -22,3 +22,4 @@ jobs:
         with:
           repo-checkout: false
           cache: false
+          go-version-input: 1.22
```
[the remark this answers is not in the document]
Comment (hairyhum) [reply .github/workflows/govulncheck.yaml:25]: It is technically using the one from `go.mod` (you can see that in the logs of a failing build) but still complains.

[the remark this answers is not in the document]
Comment (julio-lopez) [reply .github/workflows/govulncheck.yaml:25]: @hairyhum My apologies, I realize my question was very ambiguous. It meant whether `govulncheck-action` supports getting the version from the `go.mod` file, the same way that `actions/setup-go` does. It appears that it does according to the documentation.
This way, whenever the Go version is updated in the `go.mod` file, it does not need to be updated elsewhere and the Go version that is used to build remains consistent.

https://github.com/golang/govulncheck-action#using-the-govulncheck-github-action
https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

```suggestion
          go-version-file: go.mod
```

[the remark this answers is not in the document]
Comment (julio-lopez) [reply .github/workflows/govulncheck.yaml:25]: Committing this suggestion so it goes through CI.

**Review by julio-lopez — APPROVED** [review]

**Comment by julio-lopez** [comment]
Successful run: https://github.com/kanisterio/kanister/actions/runs/9183409079/job/25254066170?pr=2889

**Pull request merged**